pkg/catalog/manager/catalog_sync.go

```diff
@@ -43,6 +47,45 @@ func (m *Manager) Sync(key string, obj *v3.Catalog) (runtime.Object, error) {
 	}
 	logrus.Debugf("Chart hash comparison for global catalog %v: new -- %v --- current -- %v", catalog.Name, commit, catalog.Status.Commit)
 
+	// ensure the system catalog image cache exists
+	var systemCatalogImageCache *v1.ConfigMap
+	if catalog.Name == utils.SystemLibraryName {
+		systemCatalogImageCacheName := utils.GetCatalogImageCacheName(catalog.Name)
+		systemCatalogImageCache, err = m.ConfigMapLister.Get(namespace.System, systemCatalogImageCacheName)
+
+		// if the cache does not exist generate it
+		if err != nil && errors.IsNotFound(err) {
+			systemCatalogImageCache = &v1.ConfigMap{}
+			systemCatalogImageCache.Name = systemCatalogImageCacheName
+			systemCatalogImageCache.Namespace = namespace.System
+			err = image.CreateCatalogImageListConfigMap(systemCatalogImageCache, catalog)
+			if err != nil {
+				return nil, err
+			}
+
+			systemCatalogImageCache, err = m.ConfigMap.Create(systemCatalogImageCache)
+			if err != nil {
```

Comment:
we're listing from Lister so that might give NotFound, but the configmap might exist and Create will fail with AlreadyExists  - add check for `err != nil && !errors.IsAlreadyExists`